Updates for uncaught error handling (#3612)
cypress-io/cypress-documentation: source/api/events/catalog-of-events.md

})
```

### To conditionally turn off uncaught exception handling for a certain error

```javascript
// likely want to do this in a support file
// so it's applied to all spec files
// cypress/support/index.js

Cypress.on('uncaught:exception', (err, runnable) => {

>   // we expect a 3rd party library to error and don't want
>   // it to fail the test so we return false
>   if (err.message.includes('3rd party error')) {
>     return false
>   }

Reviewer: I kind of feel like people will interpret this as a way to turn off all 3rd party error catching. 😅 

```suggestion
  // we expect a 3rd party library error with message 'list not defined'
  // and don't want to fail the test so we return false
  if (err.message.includes('list not defined')) {
    return false
  }
```

  // we still want to ensure there are no other unexpected
  // errors, so we let them fail the test
})
```

### To conditionally turn off uncaught exception handling unhandled promise rejections

```javascript
// likely want to do this in a support file
// so it's applied to all spec files
// cypress/support/index.js

Cypress.on('uncaught:exception', (err, runnable, promise) => {

>   // when the exception originated from an unhandled promise
>   // rejection, the promise is provided as a third argument
>   // here, we choose not to fail the test in this case

Reviewer: ```suggestion
  // when the exception originated from an unhandled promise
  // rejection, the promise is provided as a third argument
  // you can turn off failing the test in this case
```
